geopmdpy/geopmdpy/session.py

```diff
@@ -41,11 +41,15 @@ def term_handler(signum, frame):
     if signum == SIGTERM and g_session_handler is not None:
         g_session_handler.stop()
 
-def _write_report(report_path, report):
+def _write_report(report_path, report, do_append=False):
     if report_path == '-':
         sys.stdout.write(report)
     else:
-        with open(report_path, "w") as fid:
+        if do_append:
+            mode = 'a'
+        else:
+            mode = 'w'
+        with open(report_path, mode) as fid:
```

Comment:
Consider keeping the report file descriptor open like we do the trace given the periodic reporting feature